app/lib/theme/CustomDiffFields.tsx

```diff
@@ -20,193 +23,161 @@ const contentSuffixElement = (
+    "diffNew",
+    "diffText",
+  ];
+
+  // Case 7 ->  111
+  if (oldData && newData && latestCommittedData) {
+    return (
+      <>
+        {latestCommittedData !== oldData && (
+          <>
+            <span
+              id={id && `${id}-${diffOldClsName}1`}
+              className={diffOldClsName}
+            >
+              {renderData(isDate, latestCommittedData)}
+            </span>
+            {contentSuffix && contentSuffixElement(id, contentSuffix)}
+            {renderArrow()}
+          </>
+        )}
+        <span id={id && `${id}-${diffOldClsName}`} className={diffOldClsName}>
+          {renderData(isDate, oldData)}
+        </span>
+        {contentSuffix && contentSuffixElement(id, contentSuffix)}
+        {renderArrow()}
+        <span id={id && `${id}-${diffNewClsName}`} className={diffNewClsName}>
+          {renderData(isDate, newData)}
+        </span>
+        {contentSuffix && contentSuffixElement(id, contentSuffix)}
+        {renderTooltip()}
+      </>
+    );
   }
-  if (latestCommittedData.isArray && latestCommittedData.length() === 0) {
-    return `${displayString}none`;
+  // case 6 ->  110
+  if (oldData && newData && !latestCommittedData) {
+    return (
+      <>
+        <span id={id && `${id}-${diffOldClsName}`} className={diffOldClsName}>
+          {renderData(isDate, oldData)}
+        </span>
+        {contentSuffix && contentSuffixElement(id, contentSuffix)}
+        {renderArrow()}
+        <span id={id && `${id}-${diffNewClsName}`} className={diffNewClsName}>
+          {renderData(isDate, newData)}
+        </span>
+        {contentSuffix && contentSuffixElement(id, contentSuffix)}
+        {renderTooltip()}
+      </>
+    );
   }
-  if (isDate) {
-    return `${displayString}${getLocaleFormattedDate(latestCommittedData)}`;
+
+  // Case 5 ->  101
+  if (oldData && !newData && latestCommittedData) {
+    return (
+      <>
+        <span id={id && `${id}-${diffOldClsName}`} className={diffOldClsName}>
+          {isDate ? getLocaleFormattedDate(oldData) : oldData}
+        </span>
+        {contentSuffix && contentSuffixElement(id, contentSuffix)}
+      </>
+    );
   }
-  if (isNumber) {
-    const decimalScale = isMoney ? 2 : numberOfDecimalPlaces ?? 0;
+  // Case 4 ->  100
+  if (oldData && !newData && !latestCommittedData) {
     return (
       <>
-        {displayString}
-        <NumberFormat
-          thousandSeparator
-          fixedDecimalScale={true}
-          decimalScale={decimalScale}
-          prefix={isMoney ? "$" : ""}
-          suffix={isPercentage ? " %" : ""}
-          displayType="text"
-          value={latestCommittedData}
-        />
+        <span id={id && `${id}-${diffNewClsName}`} className={diffNewClsName}>
+          {isDate ? getLocaleFormattedDate(oldData) : oldData}
+        </span>
+        {contentSuffix && contentSuffixElement(id, contentSuffix)}
+      </>
+    );
+  }
+  // Case 3 ->  011
+  if (!oldData && newData && latestCommittedData) {
```

Comment:
I think there's supposed to be not entered/not calculated text if there's no old data.